crates/harp/src/parse.rs

```diff
@@ -43,60 +50,57 @@ pub fn parse(code: &str) -> crate::Result<RObject> {
 
 /// Returns an EXPRSXP vector
 pub fn parse_exprs(code: &str) -> crate::Result<RObject> {
-    match unsafe { parse_status(code)? } {
-        ParseResult::Complete(x) => {
-            return Ok(RObject::from(x));
-        },
-        ParseResult::Incomplete => {
-            return Err(crate::Error::ParseError {
-                code: code.to_string(),
-                message: String::from("Incomplete code"),
-            });
-        },
-    };
+    parse_exprs_ext(code, Default::default())
 }
 
-/// This uses the R-level function `parse()` to create the srcrefs
+/// Same but creates srcrefs
 pub fn parse_exprs_with_srcrefs(code: &str) -> crate::Result<RObject> {
-    unsafe {
-        let mut protect = RProtect::new();
-
-        // Because `parse(text =)` doesn't allow `\r\n` even on Windows
-        let code = convert_line_endings(code, LineEnding::Posix);
-        let code = r_string!(code, protect);
+    parse_exprs_ext(code, RParseOptions { srcref: true })
+}
 
-        RFunction::new("base", "parse")
-            .param("text", code)
-            .param("keep.source", true)
-            .call()
```

Comment:
I also went through and confirmed that it looks like `parse(text = code, keep.source = TRUE)` is doing pretty much the exact same thing as our lower level code, i.e.:
- Calling `srcfilecopy()` to get the srcfile https://github.com/wch/r-source/blob/3052a2df21c794ead804f3960d10618ec10e90e1/src/library/base/R/parse.R#L27-L29
- Handing off to C level `R_ParseVector()` without much extra really https://github.com/wch/r-source/blob/3052a2df21c794ead804f3960d10618ec10e90e1/src/main/source.c#L286

The only difference I see between this and calling `parse()` directly is that `parse()` also fiddles with these global variables around encoding, which I think affect how `R_ParseVector()` works, but hopefully that doesn't really matter
https://github.com/wch/r-source/blob/3052a2df21c794ead804f3960d10618ec10e90e1/src/main/source.c#L240-L255